sklearn/linear_model/coordinate_descent.py

```diff
@@ -2159,3 +2161,315 @@ def __init__(self, eps=1e-3, n_alphas=100, alphas=None, fit_intercept=True,
+        If set to 'random', a random coefficient is updated every iteration
+        rather than looping over features sequentially by default. This
+        (setting to 'random') often leads to significantly faster convergence
+        especially when tol is higher than 1e-4.
+
+    random_state : int, RandomState instance, or None, optional (default=None)
+        The seed of the pseudo random number generator that selects
+        a random feature to update. Useful only when selection is set to
+        'random'.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,) | (n_targets, n_features)
+        parameter vector (w in the cost function formula)
+
+    sparse_coef_ : scipy.sparse matrix, shape (n_features, 1) | \
+            (n_targets, n_features)
+        ``sparse_coef_`` is a readonly property derived from ``coef_``
+
+    intercept_ : float | array, shape (n_targets,)
+        independent term in decision function.
+
+    n_iter_ : int
+        number of lasso iterations run by the coordinate descent solver to
+        reach the specified tolerance.
+
+    loss_ : function
+        The model loss function.
+
+    train_score_ : array, shape = [max_lasso_iterations]
+        The i-th score ``train_score_[i]`` is the loss of the
+        model at iteration ``i``.
+
+    See also
+    --------
+    Lasso
+
+    Notes
+    -----
+    The algorithm used to fit the model is coordinate descent.
+
+    Gasso, G., Rakotomamonjy, A., & Canu, S.
+    Recovering Sparse Signals With a Certain Family of Nonconvex
+    Penalties and DC Programming
+    IEEE Trans. Signal Process., 2009, 4686-4698.
+
+    Zou, Hui.
+    The Adaptive Lasso and Its Oracle Properties
+    Journal of the American Statistical Association, 2006.
+    """
+    def __init__(self, max_lasso_iterations=30, penalty='log', q=None,
+                 alpha=1.0, eps=1e-3, ada_tol=1e-4, fit_intercept=True,
+                 normalize=False, precompute=False, copy_X=True,
+                 max_iter=1000, tol=1e-4, positive=False,
+                 random_state=None, selection='cyclic'):
+        super(AdaptiveLasso, self).__init__(
+            alpha=alpha, fit_intercept=fit_intercept,
+            normalize=normalize, precompute=precompute, copy_X=copy_X,
+            max_iter=max_iter, tol=tol, warm_start=True,
+            positive=positive, random_state=random_state,
+            selection=selection)
+        self.max_lasso_iterations = max_lasso_iterations
+        self.penalty = penalty
+        self.q = q
+        self.eps = eps
+        self.ada_tol = ada_tol
+
+    def _check_params(self):
+        """Check validity of parameters and raise ValueError if not valid."""
+        if not isinstance(self.max_lasso_iterations, numbers.Number) \
+                or self.max_lasso_iterations <= 0:
+            raise ValueError("Maximum number of Lasso iterations must be"
+                             " positive; got (max_iter=%r)" % self.max_iter)
+
```

Comment:
ValueError raised for `max_iter` parameter value but conditioned on `max_lasso_iterations`